First, let's load a gro file to determine the molecular volume of the polymer matrix.


In [1]:
import time
import pickle 
from functions import *


file_path_gro = os.getcwd() + '/trial_gro/test.gro'
res1 = compute_molecular_volumes_from_gro(
                    file_path_gro,
                    prefix="test",
                    n_samples=500_000
                )

Now let's examine the results

In [2]:
print(res1)

{'testn_atoms': 10560, 'testaabb_lo_nm': [-2.247, -1.137, -4.6982], 'testaabb_hi_nm': [6.909, 6.695, 4.6160000000000005], 'testaabb_volume_nm3': 667.9193446463998, 'testhit_fraction': 0.08727, 'testunion_volume_nm3': 58.289321207291316, 'testunion_volume_stderr_nm3': 0.26658927305445257, 'testsum_sphere_volumes_nm3': 117.59011461676067, 'testpacking_fraction_vs_naive': 0.4956991614240936, 'testtotal_mass_kg': 1.0982701440542218e-22, 'testbulk_volume_m3': None}


The testuniontestunion_volume_nm3, the volume of the polymer matrix determined by the sum of the spheres taking into account the overlap os spheres -> 58.29 nm^3
is ~50% lower than the sum of the total spheres without taking into account overlaps (testsum_sphere_volumes_nm3': 117.59011461676067)
